build-caching-maven-samples/jacoco-project/pom.xml

```diff
@@ -0,0 +1,140 @@
+                                            <id>default-report</id>
+                                            <inputs>
+                                                <fileSets>
+                                                    <fileSet>
+                                                        <name>dataFile</name>
+                                                    </fileSet>
+                                                </fileSets>
+                                                <properties>
+                                                    <property>
+                                                        <name>excludes</name>
+                                                    </property>
+                                                    <property>
+                                                        <name>footer</name>
+                                                    </property>
+                                                    <property>
+                                                        <name>includes</name>
+                                                    </property>
+                                                    <property>
+                                                        <name>outputEncoding</name>
+                                                    </property>
+                                                    <property>
+                                                        <name>skip</name>
+                                                    </property>
+                                                    <property>
+                                                        <name>sourceEncoding</name>
+                                                    </property>
+                                                    <property>
+                                                        <name>title</name>
+                                                    </property>
+                                                </properties>
+                                            </inputs>
+                                            <nestedProperties>
+                                                <property>
+                                                    <name>project</name>
+                                                    <nestedProperties>
+                                                        <property>
+                                                            <name>artifactId</name>
+                                                        </property>
+                                                        <property>
+                                                            <name>build</name>
+                                                            <nestedProperties>
+                                                                <property>
+                                                                    <name>outputDirectory</name>
```

Comment:
does this mean the outputDirectory is both an input and an output?

Reply:
So this `outputDirectory` is under the [MavenProject](https://maven.apache.org/ref/3.9.2/apidocs/org/apache/maven/project/MavenProject.html) parameter in jacoco's [AbstractReportMojo](https://github.com/jacoco/jacoco/blob/master/jacoco-maven-plugin/src/org/jacoco/maven/AbstractReportMojo.java#L99). Jacoco [does some processing on this output directory](https://github.com/jacoco/jacoco/blob/master/jacoco-maven-plugin/src/org/jacoco/maven/ReportSupport.java#L158-L159), so I thought it correct to define it as an input.

The `outputDirectory` listed under `outputs` is jacoco's [ReportMojo's outputDirectory property](https://github.com/jacoco/jacoco/blob/master/jacoco-maven-plugin/src/org/jacoco/maven/ReportMojo.java#L40-L41) and is where the report is written to.

Reply:
If jacoco does some processing on the output directory then it is both an input and an output and cannot be cached since this means the goal is not reproducible.